xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-ui/src/test/java/org/xwiki/ratings/RatingsTest.java

```diff
@@ -0,0 +1,133 @@
+import org.xwiki.rendering.macro.descriptor.MacroDescriptor;
+import org.xwiki.rendering.syntax.Syntax;
+import org.xwiki.script.service.ScriptService;
+import org.xwiki.template.TemplateManager;
+import org.xwiki.test.junit5.mockito.MockComponent;
+import org.xwiki.test.page.PageTest;
+import org.xwiki.test.page.XHTML10ComponentList;
+import org.xwiki.test.page.XWikiSyntax21ComponentList;
+import org.xwiki.velocity.VelocityConfiguration;
+import org.xwiki.velocity.tools.EscapeTool;
+import org.xwiki.xml.html.filter.HTMLFilter;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Page test of the ratings.
+ *
+ * @version $Id$
+ * @since 13.0RC1
+ */
+@XWikiSyntax21ComponentList
+@XHTML10ComponentList
+class RatingsTest extends PageTest
+{
+    @MockComponent
+    @Named("template")
+    private Macro templateMacro;
+
+    // Needs to be registered for the ratings macro to be loaded.
+    @MockComponent
+    @Named("controlcharacters")
+    private HTMLFilter htmlFilter;
+
+    @BeforeEach
+    void setUp() throws Exception
+    {
+        // The Ratings macro uses Velocity macros from the macros.vm template. We need to overwrite the Velocity
+        // configuration in order to use the ClasspathResourceLoader for loading the macros.vm template from the class
+        // path.
+        VelocityConfiguration velocityConfiguration = this.oldcore.getMocker().getInstance(VelocityConfiguration.class);
+        Properties velocityConfigProps = velocityConfiguration.getProperties();
+        velocityConfigProps.put(RuntimeConstants.RESOURCE_LOADERS, "class");
+        velocityConfigProps.put(RuntimeConstants.RESOURCE_LOADER + ".class." + RuntimeConstants.RESOURCE_LOADER_CLASS,
+            ClasspathResourceLoader.class.getName());
+        velocityConfigProps.put(RuntimeConstants.VM_LIBRARY, "/templates/macros.vm");
+        velocityConfiguration = this.oldcore.getMocker().registerMockComponent(VelocityConfiguration.class);
+        when(velocityConfiguration.getProperties()).thenReturn(velocityConfigProps);
+
+        // The Ratings macro page includes the rating_macros.vm template.
+        this.oldcore.getMocker().registerMockComponent(TemplateManager.class);
+
+        MacroDescriptor templateMacroDescriptor = mock(MacroDescriptor.class);
+        when(this.templateMacro.getDescriptor()).thenReturn(templateMacroDescriptor);
+        when(templateMacroDescriptor.getParametersBeanClass()).thenReturn((Class) "".getClass());
+
+        registerVelocityTool("escapetool", new EscapeTool());
+
+        // Initialize a document reference to be used as parameter when calling displayFullRating.
+        ModelScriptService modelScriptService = mock(ModelScriptService.class);
+        when(modelScriptService.createDocumentReference("a", "b", "c\"d"))
```

Comment:
I'm curious: Why do you need a setup() since there's only a single test?

Also the `"c\"d"` seems very related to an escape test and should be in that test IMO.

WDYT?

Reply:
I've set on the setup what's generic and could be useful to future tests, but that might be (too) early optimization.
And indeed, the call to createDocumentReference is specific to my test, I'll remove the setup for now.

Reply:
And actually, having a second look help I found out I could drastically simplify the test setup.